python/Ganga/GPIDev/Base/Proxy.py

```diff
@@ -561,16 +570,25 @@ def GPIProxyObjectFactory(_obj):
         from Ganga.Core.exceptions import GangaException
         raise GangaException("%s is NOT a Proxyable object" % type(_obj))
 
-    if hasattr(_obj, proxyObject):
-        return getattr(_obj, proxyObject)
+    ## This is defined within Objects.py, we could probably store this elsehere
+    ## (We probably do) but as this is guaranteed to be accessible for GangaObjects I will use this
+    this_class = getattr(stripProxy(_obj).__class__, proxyClass)
 
-    if not hasattr(_obj.__class__, proxyClass):
-        from Ganga.Core.exceptions import GangaException
-        raise GangaException("%s is NOT a Proxyable object" % type(_obj))
+    ## Would LOVE to avoid having to perform all of these checks to get or wrap a class in a proxy here but
+    ## multiple different BAD states are currently passed to this function
+    ## In future I would like to see this fixed elsewhere and reduce the logic in this section of code
+    ## As of Ganga 6.1.15 this is needed to put objects into a semi-coherent state
 
-    proxy_class = getattr(_obj.__class__, proxyClass)
+    if isType(_obj, this_class):
+        if isProxy(_obj):
+            return _obj
+        elif not isProxy(_obj):
+            pass
+    else:
+        pass
 
-    return proxy_class(_existing=_obj)
+    proxy_class = this_class(_proxy_impl_obj_to_wrap=_obj)
+    return proxy_class
```

Comment:
The code already does that with (line 564):
```python
if hasattr(_obj, proxyObject):
  return getattr(_obj, proxyObject)
```
as for `getattr(_obj.__class__, proxyClass)` it could just be replaced with `getattr(type(_obj), proxyClass)`.

Currently,:

1. if a `GangaObject` instance is passed then if it already has a proxy object assigned, it returns that.
2. Otherwise, if the object class does not define a `_proxyClass` then it raises an error. 
3. Finally it uses the `GangaObject` instance's type's `_proxyClass` to make a new proxy object which wraps the existing `GangaObject` instance.

Which of those are you changing?